fh-sync.d.ts

```diff
@@ -0,0 +1,407 @@
+
+      type StandardCb<T> = (err: Error | null | string | undefined, res?: T | undefined) => void;
+      type NoRespCb = (err: Error | string | undefined) => void;
+
+      /**
+       * Options used to initialize sync for specific dataset
+       */
+      interface SyncInitOptions {
+          /**
+           * Value indicating how often the dataset client should be sync with the backend. Matches the clients default
+           * frequency. Value in seconds
+           */
+          syncFrequency?: number,
+
+          /**
+           * Value that will be used to decide if the dataset client is not active anymore.
+           */
+          clientSyncTimeout?: number,
+
+          /**
+           * Value that determines how long it should wait for the backend list operation to complete
+           */
+          backendListTimeout?: number,
+
+          /**
+           * Specify the max wait time the dataset can be scheduled to sync again after its previous schedule, in seconds.
+           */
+          maxScheduleWaitTime?: number
+      }
+
+      /**
+       * Parameters object for request and response interceptors
+       */
+      interface SyncInterceptParams {
+          query_params: any;
+          metaData: any;
+      }
+      /**
+       * Connect sync server to mongo and redis. Returns the MongoDB and Redis clients being used internally.
+       *
+       * @param mongoDBConnectionUrl - Unique id of the dataset (usually collection, table in your database)
+       * @param mongoDBConnectionOption - Connection options for the MongoDB driver
+       * @param redisUrl - Redis connection URL
+       * @param callback - Callback that will be invoked once connections are setup
+       */
+      function connect(mongoDBConnectionUrl: string, mongoDBConnectionOption: any, redisUrl: string, callback: (err: any, mongoDbClient?: any, redisClient?: any) => void): void;
+
+      /**
+       * Initialize sync for specific dataset. The passed datasetId must be a unique string.
+       *
+       * @param datasetId - Unique name of the dataset to initialise
+       * @param options - Specific options to apply to this dataset
+       * @param callback - Callback that will be invoked once initialisation is complete
+       */
+      function init(datasetId: string, options: SyncInitOptions, callback: StandardCb<void>): void;
+
+      /**
+       * Internal method used to invoke sync methods. Should be used to handle json request from client.
+       *
+       * Supported operations are 'sync', 'syncRecords', 'listCollisions', 'removeCollision' and should be passed as a
+       * "fn" key in the options object.
+       *
+       * @param datasetId - The dataset to invoke the operation on
+       * @param options - Options to pass to the invocation
+       * @param callback - Function that will receive the invocation results
+       */
+      function invoke(datasetId: string, options: InvokeOptions, callback: (err: any, result: any) => void): void;
```

Comment:
Probably should be `StandardCb<any>` for consistency, though it doesn't matter too much here. Mostly just narrows the type of err a little bit.